Ignore primary_key_prefix_type on HABTM join models

mlarraz opened the pull request:
### Summary
HABTM join models do not have primary keys by default,
but setting `primary_key_prefix_type` on `ActiveRecord::Base`
will cause all models to expect primary keys to be present.

Overriding the value to `nil` will cause the model to check
the db for its primary key instead.

Resolves https://github.com/rails/rails/issues/28502

rafaelfranca commented:
Could add a test for this behavior so the bug can't be introduce again?

rafaelfranca commented on activerecord/test/cases/associations/has_and_belongs_to_many_associations_test.rb:
```diff
@@ -168,6 +168,21 @@ def test_proper_usage_of_primary_keys_and_join_table
+
+    ActiveRecord::Base.primary_key_prefix_type = :table_name_with_underscore
+
+    load 'models/country.rb'
```
Why do you need to load the models again?

mlarraz replied:
I need to reset the join model's primary key definition, but I want to avoid directly referencing the model as it's a private constant. 

The more direct way to do it is
```ruby
Country.const_get(:HABTM_Treaties).reset_primary_key
```

I'm not a fan of either approach, so open to suggestions

mlarraz committed:
Add test for HABTM join table ignoring primary key prefix

mlarraz committed:
Ignore primary_key_prefix_type on HABTM join models

HABTM join models do not have primary keys by default,
but setting primary_key_prefix_type on ActiveRecord::Base
will cause all models to expect primary keys to be present.

Overriding the value to `nil` will cause the model to check
the db for its primary key instead.
